crates/server/src/authority/authority.rs

```diff
@@ -129,7 +129,7 @@ pub trait Authority: Send + Sync {
         name: &LowerName,
         rtype: RecordType,
         lookup_options: LookupOptions,
-    ) -> Result<Self::Lookup, LookupError>;
+    ) -> Result<Option<Self::Lookup>, LookupError>;
```

Comment:
I think this PR is currently too large for me to review it. If you want me to digest these changes, I suggest you split this trait API change out into a separate PR and land it separately, with a clear rationale (which can refer to the blocklist).

Reply:
I think I'll split this up into 3 PRs, one for the chained authority implementation, one for the lookup call changes, and one for the blocklist itself.

@bluejekyll if you have some time it would be nice to get some feedback on these lookup call changes in particular.  All of this stuff is based on your suggestion [here](https://github.com/hickory-dns/hickory-dns/pull/2113#discussion_r1439074772) to use an Option to represent the lookup failure in the blocklist.  All of the lookup call sites had to change as a result, and I think using Option, compared to the LookupErr::NotHandled solution, adds more complexity to the lookup call semantics than it adds clarity to the situation where an authority needs to signal that it didn't handle a lookup and the next authority in the chain should be consulted.

Reply:
Thanks, that's good context! Let's see what @bluejekyll thinks.